### DISCUSSION TITLE
Mono is not active anymore, can we remove this dependency to run NuGet on Linux?

### DISCUSSION THREAD

**Discussion by akirayamamoto** [NuGet/Home#12734] [General]
Mono is not active anymore yet the [official documentation](https://learn.microsoft.com/en-us/nuget/reference/nuget-exe-cli-reference#macoslinux) recommends using Mono to run NuGet on Linux.

Is it time to provide an alternative?

**Comment by zivkan** [comment]
.NET (formerly known as .NET Core) has the `dotnet` CLI that works on Linux, including `dotnet restore`, `dotnet pack`, and some `dotnet nuget` commands (plus some commands that never existed in nuget.exe).

We also have this issue that lists the missing features in the dotnet CLI that the nuget.exe CLI has: https://github.com/NuGet/Home/issues/4855#issuecomment-1205936210

You didn't say what feature or command you want to run, but all the most important commands are there already, and other missing commands get implemented based on prioritization (so upvoting issues helps), as well as our team's capacity.

**Reply by akirayamamoto** [reply]
Thanks for the quick reply @zivkan. The command we are interested is `nuget install`. We are using it to download a NuGet package. I see that according to the link you mentioned that is only partially implemented using `dotnet`. How would I go to do the same in an empty directory using `dotnet`? I can see that I could do `dotnet new` and `dotnet add package` but they are not exactly the same.

We are considering using the NuGet server REST API and downloading the package directly from there as an alternative.

**Reply by zivkan** [reply]
A replacement for `nuget install` has been requested: https://github.com/NuGet/Home/issues/12513. We prioritize work on a number of factors, one of which is upvotes. At this time, that issue only has 1 upvote.

While not exactly the same use case as `nuget install`, years ago NuGet's client repo used to have a repo `packages.config` with build tools and libraries. We converted it to a csproj, using `PackageDownload` items to avoid TFM compatibility checks, and set the `RestorePackagesPath` property to set the "global packages folder" to a specific location that the rest of our build could hardcode: https://github.com/NuGet/NuGet.Client/pull/3073/files.  You might be able to use a similar technique and therefore be able to use `dotnet restore bootstrap.proj` as we did. But it depends on how you use `nuget install`.
